### ISSUE TITLE
Dependencies can't be declared final

### ISSUE THREAD

**Issue by dingari** [boost-ext/sml#483]
## Expected Behavior
I have an FSM that has dependencies of classes that are declared final.

## Actual Behavior
The program can't be compiled because SML tries to inherit from my dependencies. This is very inconvenient, since some of my dependencies are from external frameworks that I don't control. I suppose I could wrap them in another struct, but that seems like an unnecessary inconvenience. This was not an issue with previous versions.

## Steps to Reproduce the Problem

See [example](https://godbolt.org/z/an4acGE7h)

## Specifications

  - Version: 1.1.4
  - Platform: MSVC

**Comment by tralamazza** [comment]
seems to be msvc specific?

**Comment by dingari** [comment]
Yes, seems to be MSVC specific. Do you think it's a compiler bug? It's strange that it does not happen with Clang/GCC, since they clearly don't allow you to inherit from a final class either.
